meson.build

```diff
@@ -50,6 +50,13 @@ m_dep = cc.find_library('m', required : false)
 if m_dep.found()
   add_project_link_arguments('-lm', language : 'c')
 endif
+if host_machine.system() == 'os400'
+  add_global_arguments('-pthread', language : 'cpp')
+  add_global_arguments('-D__STDC_FORMAT_MACROS', language : 'cpp')
```

Comment:
These two lines should use `add_project_argument` rather than `add_global_argument`.

Reply:
Adding `__STDC_FORMAT_MACROS` is fine - I guess the OS/400 C++ compiler is a bit outdated (modern C++ compilers should unconditionally define what's behind this macro - see https://stackoverflow.com/questions/8132399/how-to-printf-uint64-t-fails-with-spurious-trailing-in-format/8132440#8132440).

Reply:
> These two lines should use `add_project_argument` rather than `add_global_argument`.

Yeah. I would try add_project_argument.  BTW, in fact, by checking the build.ninja , we can find that "-pthread" could be added for some c++ source code, while, it only failed on some files , such as control.cc. To be honest , I didn't figure out why mason treat control.cc specially.  



> Adding `__STDC_FORMAT_MACROS` is fine - I guess the OS/400 C++ compiler is a bit outdated (modern C++ compilers should unconditionally define what's behind this macro - see https://stackoverflow.com/questions/8132399/how-to-printf-uint64-t-fails-with-spurious-trailing-in-format/8132440#8132440).

Yes, this is exactly.